File: SortingAlgorithms/python/insertion sort.py
def insertion_sort(array):         #function to sort array using insertion sort
    comparison=0
    shift=0
    for i in range(1,len(array)):
        j=i-1
        val=array[i]
        while(j>=0 and val<array[j]):
            shift+=1
            comparison+=1
            array[j+1]=array[j]
            array[j]=val
            j-=1
        comparison+=1
        print("Array after ",i+1,"th pass is : ",array)
    print("Array after sorting is : ",array)
    print(shift," number of shift and",comparison," comparisons")

def input_array():                       #function for taking input of array
    attended=[]
    number_of_attendee=int(input("Enter total : "))
    for i in range(number_of_attendee):
        tep=int(input("Enter number :"))
        attended.append(tep)
    return attended

if __name__ == '__main__':
    arr=input_array()
    insertion_sort(arr)
    print("Array after sorting using selection sort : ",arr)
    

File: SortingAlgorithms/java/InsertionSort.java
class InsertionSort {
    void sort(int arr[])
    {
        for (int i = 1; i < arr.length; ++i) {
            int key = arr[i];
            int j = i - 1;

            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }

    public static void main(String args[])
    {
        int arr[] = { 2, 1, 3, 4, 5 };

        new InsertionSort().sort(arr);

        for (int i = 0; i < arr.length; ++i)
            System.out.print(arr[i] + " ");

        System.out.println();
    }
}